init.lua

```diff
@@ -166,5 +166,60 @@ function M.text(txt, opts)
   return win
 end
 
+function M.audio(data, opts)
+  opts = opts or {}
+  local win = opts.win or uid()
+
+  if not pcall(require, 'audio') then
+      print("Warning: audio package could not be loaded. Skipping audio.")
+      return
+  end
+
+  local fname
+  local delete = false
+  if torch.isTensor(data) then -- audio as tensor
+      if data:dim() ~= 2 then
+          print('Warning: audio tensor has to be 2D Tensor of NSamples x NChannels. Other tensor shapes are not supported')
+          return
+      end
+      local sampleRate = opts.rate or 44100 -- default sample rate
+      -- use temporary file
+      fname = os.tmpname() .. '.ogg'
+      audio.save(fname, data, sampleRate)
```

Comment:
I've added audio.compress / decompress to the audio package today. However, it is only supported on Linux OOB because it uses fmemopen.
I added a commit here to use that path where possible, and fallback to temporary files.